src/SymbolsAuth/SymwebHandler.cs

```diff
@@ -0,0 +1,162 @@
+﻿using Azure.Core;
+using System;
+using System.IO;
+using System.Net.Http;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.Diagnostics.Symbols.Authentication
+{
+    /// <summary>
+    /// Contains constants, static properties and helper methods pertinent to Symweb.
+    /// </summary>
+    internal static class Symweb
+    {
+        /// <summary>
+        /// The OAuth scope to use when requesting tokens for Symweb.
+        /// </summary>
+        public const string Scope = "af9e1c69-e5e9-4331-8cc5-cdf93d57bafa/.default";
+
+        /// <summary>
+        /// The url host for Symweb.
+        /// </summary>
+        public const string SymwebHost = "symweb.azurefd.net";
+
+        /// <summary>
+        /// Try to find the authority endpoint for Symweb given a full URI.
+        /// </summary>
+        /// <param name="requestUri">The request URI.</param>
+        /// <param name="authority">The authority, if found.</param>
+        /// <returns>True if <paramref name="requestUri"/> represents a path to a
+        /// resource in Symweb.</returns>
+        public static bool TryGetAuthority(Uri requestUri, out Uri authority)
+        {
+            if (!requestUri.IsAbsoluteUri)
+            {
+                authority = null;
+                return false;
+            }
+
+            UriBuilder builder = null;
+            string host = requestUri.DnsSafeHost;
+            if (host.Equals(SymwebHost, StringComparison.OrdinalIgnoreCase))
+            {
+                builder = new UriBuilder
+                {
+                    Host = SymwebHost
+                };
+            }
+
+            if (builder is null)
+            {
+                // Not a Symweb URI.
+                authority = null;
+                return false;
+            }
+
+            builder.Scheme = requestUri.Scheme;
+            if (!requestUri.IsDefaultPort)
+            {
+                builder.Port = requestUri.Port;
+            }
+
+            authority = builder.Uri;
+            return true;
+        }
+    }
+
+    /// <summary>
+    /// A handler that adds authorization for Symweb.
+    /// </summary>
+    public sealed class SymwebHandler : SymbolReaderAuthHandler
```

Comment:
There's a lot of similarity between this and the AzureDevOpsHandler. I wonder if there could be a common base class.

Reply:
You're right.  I'm not sure.  Let me take this as a follow-up to this PR.